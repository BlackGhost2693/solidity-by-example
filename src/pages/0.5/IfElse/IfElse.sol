pragma solidity ^0.5.16;

contract IfElse {
    function foo(uint x) public pure returns (uint) {
        if (x < 10) {
            return 0;
        } else if (x < 20) {
            return 1;
        } else {
            return 2;
        }
    }
}
