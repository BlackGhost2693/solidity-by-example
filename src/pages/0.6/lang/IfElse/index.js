import React from "react"
import Example from "../../../components/Example"
import html from "./index.html.js"
import { SOL_VERSION } from "../constants"

export default function (props) {
  return (
    <Example
      title="If / Else"
      version={SOL_VERSION}
      description="If / Else conditional statement in Solidity"
      html={html}
    />
  )
}
