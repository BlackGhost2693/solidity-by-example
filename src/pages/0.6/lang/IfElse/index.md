Solidity support conditional statements `if`, `else if` and `else`.

```solidity
{{{IfElse}}}
```
