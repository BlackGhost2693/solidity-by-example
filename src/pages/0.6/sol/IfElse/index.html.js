// metadata
export const version = "0.6.0"
export const title = "If / Else"
export const description = "If / Else conditional statement in Solidity"

const html = `<p>Solidity support conditional statements <code>if</code>, <code>else if</code> and <code>else</code>.</p>
<pre><code class="language-solidity">pragma solidity ^0.6.0;

contract IfElse {
    function foo(uint x) public pure returns (uint) {
        if (x &lt; 10) {
            return 0;
        } else if (x &lt; 20) {
            return 1;
        } else {
            return 2;
        }
    }
}
</code></pre>
`

export default html
