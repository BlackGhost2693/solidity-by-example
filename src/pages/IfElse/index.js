import React from "react"
import Example from "../../components/Example"
import html from "./index.html.js"

export default function (props) {
  return (
    <Example
      title="If / Else"
      description="If / Else conditional statement in Solidity"
      html={html}
    />
  )
}
