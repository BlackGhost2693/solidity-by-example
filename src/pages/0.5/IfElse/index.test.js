import React from "react"
import { shallow } from "enzyme"
import IfElse from "./index"

test("it renders", () => {
  const component = shallow(<IfElse />)

  expect(component).toMatchSnapshot()
})
