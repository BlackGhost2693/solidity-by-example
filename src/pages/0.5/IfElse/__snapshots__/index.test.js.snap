// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<Example
  description="If / Else conditional statement in Solidity"
  html="<p>Solidity support conditional statements <code>if</code>, <code>else if</code> and <code>else</code>.</p>
<pre><code class=\\"language-solidity\\">pragma solidity ^0.5.16;

contract IfElse {
    function foo(uint x) public pure returns (uint) {
        if (x &lt; 10) {
            return 0;
        } else if (x &lt; 20) {
            return 1;
        } else {
            return 2;
        }
    }
}
</code></pre>
"
  title="If / Else"
  version="0.5.16"
/>
`;
